indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,995 @@
+        }
+        return null;
+      }
+      try {
+        loadedBytes = job.getCounters().findCounter(COUNTER_GROUP, COUNTER_LOADED).getValue();
+        writtenBytes = job.getCounters().findCounter(COUNTER_GROUP, COUNTER_WRITTEN).getValue();
+      }
+      catch (IOException ex) {
+        log.error(ex, "Could not fetch counters");
+      }
+      final JobID jobID = job.getJobID();
+
+      final Path jobDir = getJobPath(jobID, job.getWorkingDirectory());
+      final FileSystem fs = jobDir.getFileSystem(job.getConfiguration());
+      final RemoteIterator<LocatedFileStatus> it = fs.listFiles(jobDir, true);
+      final List<Path> goodPaths = new ArrayList<>();
+      while (it.hasNext()) {
+        final LocatedFileStatus locatedFileStatus = it.next();
+        if (locatedFileStatus.isFile()) {
+          final Path myPath = locatedFileStatus.getPath();
+          if (ConvertingOutputFormat.DATA_SUCCESS_KEY.equals(myPath.getName())) {
+            goodPaths.add(new Path(myPath.getParent(), ConvertingOutputFormat.DATA_FILE_KEY));
+          }
+        }
+      }
+      if (goodPaths.isEmpty()) {
+        log.warn("No good data found at [%s]", jobDir);
+        return null;
+      }
+      final List<DataSegment> returnList = ImmutableList.copyOf(
+          Lists.transform(
+              goodPaths, new Function<Path, DataSegment>()
+              {
+                @Nullable
+                @Override
+                public DataSegment apply(final Path input)
+                {
+                  try {
+                    if (!fs.exists(input)) {
+                      throw new ISE(
+                          "Somehow [%s] was found but [%s] is missing at [%s]",
+                          ConvertingOutputFormat.DATA_SUCCESS_KEY,
+                          ConvertingOutputFormat.DATA_FILE_KEY,
+                          jobDir
+                      );
+                    }
+                  }
+                  catch (final IOException e) {
+                    throw Throwables.propagate(e);
+                  }
+                  try (final InputStream stream = fs.open(input)) {
+                    return HadoopDruidConverterConfig.jsonMapper.readValue(stream, DataSegment.class);
+                  }
+                  catch (final IOException e) {
+                    throw Throwables.propagate(e);
+                  }
+                }
+              }
+          )
+      );
+      if (returnList.size() == segments.size()) {
+        return returnList;
+      } else {
+        throw new ISE(
+            "Tasks reported success but result length did not match! Expected %d found %d at path [%s]",
+            segments.size(),
+            returnList.size(),
+            jobDir
+        );
+      }
+    }
+    catch (InterruptedException | ClassNotFoundException e) {
+      runtimeException = Throwables.propagate(e);
+      throw runtimeException;
+    }
+    catch (RuntimeException ex) {
+      runtimeException = ex;
```

Comment:
why not just use "throwable" to keep reference of exception as well and variable runtimeException is probably redundant.